File: uva/11530/ad_hoc.py
def main():
    keys = [
        '', 'abc', 'def',
        'ghi', 'jkl', 'mno',
        'pqrs', 'tuv', 'wxyz',
        '', ' ', '',
    ]

    num_tests = int(input())
    for i in range(1, num_tests+1):
        text = input()
        num_pressings = 0
        for c in text:
            for key_set in keys:
                if c in key_set:
                    num_pressings += key_set.index(c) + 1
        print(f'Case #{i}: {num_pressings}')

if __name__ == '__main__':
    main()

File: uva/11530/AdHoc.java
import java.util.Scanner;

public class AdHoc {

    public static void main(String[] args) {
        String[] keys = {
            "", "abc", "def",
            "ghi", "jkl", "mno",
            "pqrs", "tuv", "wxyz",
            "", " ", "",
        };
        Scanner sc = new Scanner(System.in);
        int tc = Integer.parseInt(sc.nextLine());
        for (int i = 1; i <= tc; i++) {
            int numPressings = 0;
            var msg = sc.nextLine().split("");
            for (var c : msg)
                for (var keySet : keys)
                    if (keySet.contains(c)) {
                        numPressings += keySet.indexOf(c) + 1;
                        break;
                    }

            System.out.printf("Case #%d: %d\n", i, numPressings);
        }
        sc.close();
    }
}